src/annoylib.h

```diff
@@ -878,11 +880,25 @@ template<typename S, typename T, typename Distance, typename Random>
       if (f == NULL)
         return false;
 
-      fwrite(_nodes, _s, _n_nodes, f);
-      fclose(f);
+      bool write_failed = false;
```

Comment:
don't think this one is really needed, could just `return false` at each point?

not blocking though

Reply:
I refactored that a few times deciding if I wanted to unload or not in case of error.  That is a vestige of that.